go/tasks/pluginmachinery/k8s/plugin.go

```diff
@@ -51,6 +51,31 @@ type PluginProperties struct {
 	// override that behavior unless the resource that gets created for this plugin does not consume resources (cluster's
 	// cpu/memory... etc. or external resources) once the plugin's Plugin.GetTaskPhase() returns a terminal phase.
 	DisableDeleteResourceOnFinalize bool
+	// Optionally specifies an override for the default OnAbort behavior (deleting the underlying resource).
+	AbortOverride PluginAbortOverride
```

Comment:
Let's delete this... As discussed on the thread, I want to keep this struct serializable... from yaml -> go...
```suggestion
```